### ISSUE TITLE
Understanding how Caldera deals with dependencies for abilities

### ISSUE THREAD

**Issue by 0xd3vnull** [mitre/caldera#2489]
Hi there

Can someone please explain how the dependencies for an ability are resolved before an ability is run by Caldera. (As discussed in https://github.com/mitre/atomic/pull/33) and how to tell if the dependencies were successfully resolved?

For example,...

Dump LSASS.exe Memory using ProcDump

Has a dependency that ProcDump is installed in `PathToAtomicsFolder\T1003.001\bin\procdump.exe`, does the ability check to see if the dependency is met before running (which it doesn't look like it is) and if it isn't and you are required the run the dependency PowerShell to download the script externally, which directory should this be run from so it is in the path of the SANDCAT agent as the command contains a relative path.

Since SANDCAT looks like it is installing in C:\Users\Public, is that where you should be running the prerequisite script from? It doesn't appear to be the case, and the relative path issue is really throwing me getting this ability to work.

I have also tried running the script inside C:\Windows\temp which seemed another obvious place, but without success.

I apologise for double posting, as I also raised this in the discussion, but that area doesn't appear to get a lot of traffic.

**Comment by ArtificialErmine** [comment]
Hey @0xd3vnull - let's see if we can't get to the bottom of this. I think this is unfortunately one of the places where the meshing the atomic framework into Caldera leads into oddities due to differences in how things are processed and stored. When importing Atomic abilities, we can only really detect dependencies that they explicitly call out, in which case we make some effort to find ways to incorporate them into our framework. In this particular case, though, that doesn't work, so we have to fall back to using the test as written by the original creators (with no knowledge of whether or not it will work, as you pointed out), which expects to find the file literally at `PathToAtomicsFolder\T1003.001\bin\procdump.exe` - which will almost never be on any path you make on the target host. I would suggest either modifying the ability's command either directly to just `procdump.exe` at the relevant location, or to create the expected folder structure somewhere on the target.

To summarize - the ability references `PathToAtomicsFolder\T1003.001\bin\procdump.exe`, which either needs to exist or be manually changed. Hopefully that clears things up a bit, but if not, let me know and I can try to help further.

**Comment by 0xd3vnull** [comment]
What if there was an option to set $PathToAtomicsFolder as a variable? Is this possible?

**Comment by ArtificialErmine** [comment]
Closing due to inactivity. If the issue persists, feel free to reopen or file a new ticket..
